fsharp-backend/src/BackendOnlyStdLib/HttpBytesClient.fs

```diff
@@ -0,0 +1,212 @@
+
+// There has been quite a history of .NET's HttpClient having problems,
+// including socket exhaustion and DNS results not expiring.
+// The history is outlined well here:
+// https://www.stevejgordon.co.uk/httpclient-connection-pooling-in-dotnet-core
+//
+// As of .NET 6 it seems we no longer need to worry about either socket
+// exhaustion or DNS issues. It appears that we can use either multiple HTTP
+// clients or just one, we use just one for efficiency.
+// See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0#alternatives-to-ihttpclientfactory
+//
+// Note that the number of sockets was verified manually, with:
+// `sudo netstat -apn | grep _WAIT`
+// TODO: I don't see where "the nunber of sockets" is actually configured?
+let private socketHandler : HttpMessageHandler =
+  new SocketsHttpHandler(
+    // Avoid DNS problems
+    PooledConnectionIdleTimeout = System.TimeSpan.FromMinutes 5.0,
+    PooledConnectionLifetime = System.TimeSpan.FromMinutes 10.0,
+
+    AutomaticDecompression = System.Net.DecompressionMethods.None,
+
+    AllowAutoRedirect = false,
+
+    UseProxy = true,
+    Proxy = System.Net.WebProxy(Config.httpclientProxyUrl, false),
+
+    // Don't add a RequestId header for opentelemetry
+    ActivityHeadersPropagator = null,
+
+    // Users share the HttpClient, don't let them share cookies!
+    UseCookies = false
+  )
+
+let private httpClient : HttpClient =
+  new HttpClient(
+    socketHandler,
+    disposeHandler = false,
+
+    // These seem like reasonable defaults
+    Timeout = System.TimeSpan.FromSeconds 30.0,
+    MaxResponseContentBufferSize = 1024L * 1024L * 100L // 100MB
+  )
+
+let private httpCall
+  (url : string)
+  (queryParams : (string * string list) list)
+  (method : HttpMethod)
+  (reqHeaders : HttpHeaders.T)
+  (reqBody : byte array)
+  : Task<Result<HttpResult, ClientError>> =
+  task {
+    use _ =
+      Telemetry.child
+        "HttpClient.call"
+        [ "request.url", url; "request.method", method ]
+    try
+      let uri = System.Uri(url, System.UriKind.Absolute)
+
+      // currently we only support http(s) requests
+      if uri.Scheme <> "https" && uri.Scheme <> "http" then
+        return Error { code = None; error = "Unsupported protocol" }
+      else
+        let reqUri =
+          let queryString =
+            let query =
+              // Remove the question mark
+              if uri.Query.Length > 0 then String.dropLeft 1 uri.Query else uri.Query
+            HttpQueryEncoding.createQueryString query queryParams
+
+          System.UriBuilder(
+            Scheme = uri.Scheme,
+            Host = uri.Host,
+            Port = uri.Port,
+            Path = uri.AbsolutePath,
+            Query = queryString
+          )
+
+        use req =
+          new HttpRequestMessage(
+            method,
+            string reqUri,
+            Version = System.Net.HttpVersion.Version30,
```

Comment:
Does this say "use exactly version three" or does it allow any HTTP version?

Reply:
I'm not sure. I'll add a specific TODO there to research later.